<launch>
	<!-- Include files for working with robot -->
	<include file="$(find baxter_moveit_config)/launch/planning_context.launch">
		<arg name="load_robot_description" value="true"/>
	</include>
	<include file="$(find baxter_moveit_config)/launch/move_group.launch"/>
	<node pkg="baxter_interface" type="joint_trajectory_action_server.py" name="joint_trajectory_action_server"/> 

	<!-- Include file for streaming position from Optitrack -->
	<include file="$(find qlearn_baxter)/launch/across.launch"/>

	<!-- Start Optitrack node for broadcasting transformations -->
	<node pkg="qlearn_baxter" type="optitrack_find.py" name="Optitrack_find" output="screen">
		<remap from="bax_head/pose" to="vrpn_client_node/kruna/pose"/>
		<remap from="bax_arm/pose" to="vrpn_client_node/ruka/pose"/>
	</node>
	<node pkg="qlearn_baxter" type="baxter_find_tf.py" name="Baxter_find" output="screen"/>

	<!-- Start Baxter's client and server and qlearning algorithm -->
	<node pkg="qlearn_baxter" type="BaxterArmServer.py" name="Server" output="screen"/>
	<node pkg="qlearn_baxter" type="testing.py" name="BaxterTest" output="screen">
		<param name="left_rod" value="0.2"/>
		<param name="right_rod" value="0.2"/>
		<param name="center_rod" value="0.2"/>
		<param name="disk_height" value="0.06"/>
	</node>
</launch>